runtime/manta/src/migrations/assets_genesis.rs

```diff
@@ -0,0 +1,397 @@
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+#![allow(clippy::unnecessary_cast)]
+
+use crate::sp_api_hidden_includes_construct_runtime::hidden_include::StorageHasher;
+use codec::Encode;
+use core::marker::PhantomData;
+use cumulus_primitives_core::ParaId;
+#[allow(deprecated)]
+use frame_support::migration::{
+    clear_storage_prefix, get_storage_value, put_storage_value, storage_key_iter,
+};
+use frame_support::{
+    dispatch::GetStorageVersion,
+    pallet_prelude::Weight,
+    traits::{Currency, Get, OnRuntimeUpgrade, StorageVersion},
+    Blake2_128Concat,
+};
+use manta_primitives::{
+    assets::{AssetConfig, AssetLocation, AssetRegistryMetadata},
+    types::{Balance, MantaAssetId},
+};
+use sp_runtime::BoundedVec;
+use sp_std::vec::Vec;
+
+pub type DepositBalanceOf<T, I = ()> = <<T as pallet_assets::Config<I>>::Currency as Currency<
+    <T as frame_system::Config>::AccountId,
+>>::Balance;
+
+pub const INITIAL_PALLET_ASSETS_MANAGER_VERSION: u16 = 0;
+pub const INITIAL_PALLET_ASSETS_VERSION: u16 = 0;
+
+pub struct AssetsGenesis<T>(PhantomData<T>);
+impl<T> OnRuntimeUpgrade for AssetsGenesis<T>
+where
+    T: pallet_asset_manager::Config + pallet_assets::Config,
+{
+    fn on_runtime_upgrade() -> Weight {
+        let asset_manager_storage_version =
+            <pallet_asset_manager::Pallet<T> as GetStorageVersion>::on_chain_storage_version();
+        let assets_storage_version =
+            <pallet_assets::Pallet<T> as GetStorageVersion>::on_chain_storage_version();
+        if asset_manager_storage_version != INITIAL_PALLET_ASSETS_MANAGER_VERSION
+            || assets_storage_version != INITIAL_PALLET_ASSETS_VERSION
+        {
+            log::info!("Aborting migration due to unexpected on-chain storage versions for pallet-assets-manager: {:?} and pallet-assets: {:?}. Expectation was: {:?} and {:?}.", asset_manager_storage_version, assets_storage_version, INITIAL_PALLET_ASSETS_MANAGER_VERSION, INITIAL_PALLET_ASSETS_VERSION );
+            return T::DbWeight::get().reads(2);
+        }
+
+        log::info!(target: "asset-manager", "Starting migration for AssetManager...");
+
+        let pallet_prefix: &[u8] = b"AssetManager";
+        let _ = clear_storage_prefix(pallet_prefix, b"AssetIdLocation", b"", None, None);
+        let _ = clear_storage_prefix(pallet_prefix, b"LocationAssetId", b"", None, None);
+        let _ = clear_storage_prefix(pallet_prefix, b"AssetIdMetadata", b"", None, None);
+        let _ = clear_storage_prefix(pallet_prefix, b"UnitsPerSecond", b"", None, None);
+        let _ = clear_storage_prefix(pallet_prefix, b"MinXcmFee", b"", None, None);
+        let _ = clear_storage_prefix(pallet_prefix, b"AllowedDestParaIds", b"", None, None);
+
+        put_storage_value(
+            pallet_prefix,
+            b"NextAssetId",
+            b"",
+            <T::AssetConfig as AssetConfig<T>>::StartNonNativeAssetId::get(),
```

Comment:
this puts `8` in here, you probably wanted `StartNativeAssetId`

Reply:
ok nvm, i see the native assert is registered here manually

Reply:
Yeah better to be explicit and not use these magic numbers.

Reply:
No. We're setting the native asset with the migraiton so the next-asset-id should be the next non-native id.